azure-mgmt-search/src/main/java/com/microsoft/azure/management/search/SearchServices.java

```diff
@@ -0,0 +1,200 @@
+
+  /***********************************************************
+   * Getters
+   ***********************************************************/
+
+  /**
+   * Gets the primary and secondary admin API keys for the specified Azure Search service.
+   *
+   * @param resourceGroupName The name of the resource group within the current subscription. You can obtain this value from the Azure Resource Manager API or the portal.
+   * @param searchServiceName The name of the Azure Search service associated with the specified resource group.
+   * @throws IllegalArgumentException thrown if parameters fail the validation
+   * @throws CloudException thrown if the request is rejected by server
+   * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent
+   * @return the AdminKeys object if successful.
+   */
+  AdminKeys getAdminKeys(String resourceGroupName, String searchServiceName);
+
+  /**
+   * Gets the primary and secondary admin API keys for the specified Azure Search service.
+   *
+   * @param resourceGroupName The name of the resource group within the current subscription. You can obtain this value from the Azure Resource Manager API or the portal.
+   * @param searchServiceName The name of the Azure Search service associated with the specified resource group.
+   * @throws IllegalArgumentException thrown if parameters fail the validation
+   * @return the observable to the AdminKeys object
+   */
+  Observable<AdminKeys> getAdminKeysAsync(String resourceGroupName, String searchServiceName);
+
+  /**
+   * Returns the list of query API keys for the given Azure Search service.
+   *
+   * @param resourceGroupName The name of the resource group within the current subscription. You can obtain this value from the Azure Resource Manager API or the portal.
+   * @param searchServiceName The name of the Azure Search service associated with the specified resource group.
+   * @throws IllegalArgumentException thrown if parameters fail the validation
+   * @throws CloudException thrown if the request is rejected by server
+   * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent
+   * @return the List&lt;QueryKey&gt; object if successful.
+   */
+  List<QueryKey> listQueryKeys(String resourceGroupName, String searchServiceName);
+
+  /**
+   * Returns the list of query API keys for the given Azure Search service.
+   *
+   * @param resourceGroupName The name of the resource group within the current subscription. You can obtain this value from the Azure Resource Manager API or the portal.
+   * @param searchServiceName The name of the Azure Search service associated with the specified resource group.
+   * @throws IllegalArgumentException thrown if parameters fail the validation
+   * @return the observable to the List&lt;QueryKeyInner&gt; object
```

Comment:
we have had a bit of dilemma with using Java-specific descriptions in async methods, because when converted to C#, these don't make sense anymore.  One of the alternative wordings we've used that at least continue making sense in C# is something like
` @return a representation of the future computation of this call`